src/main/java/com/snowplowanalytics/redash/RedashClient.java

```diff
@@ -0,0 +1,422 @@
+     */
+    public User getUserById(int userId) throws IOException {
+        String url = baseUrl + USERS + "/" + userId + APIKEY_STRING + apiKey;
+        String returnValue = get(url, false);
+        return resultResolver(User.class, returnValue);
+    }
+
+    /**
+     * Provides opportunity to retrieve the user-group from redash server by it's id. Implemented as util method.
+     *
+     * @return DataSource instance
+     * @throws IllegalArgumentException if there is no user-group with such id
+     * @throws IOException              if server is not available due to connection error or API key is invalid.
+     */
+    public DataSource getDataSourceById(int id) throws IOException {
+        String url = baseUrl + DATA_SOURCES + "/" + id + APIKEY_STRING + apiKey;
+        String returnValue = get(url, false);
+        if (MESSAGE_INTERNAL_SERVER_ERROR.equals(returnValue)) {
+            throw new IllegalArgumentException(MESSAGE_INTERNAL_SERVER_ERROR);
+        }
+        return new Gson().fromJson(returnValue, DataSource.class);
+    }
+
+    private boolean isDataSourceAlreadyUpToDate(DataSource fromDataBase, DataSource whichToUpdate) {
+        return fromDataBase.getHost().equals(whichToUpdate.getHost())
+                && fromDataBase.getPort() == whichToUpdate.getPort()
+                && fromDataBase.getUser().equals(whichToUpdate.getUser())
+                && fromDataBase.getDbName().equals(whichToUpdate.getDbName());
+    }
+
+    private boolean isEntityAlreadyExists(List<? extends BaseEntity> list, String name) throws IOException {
+        return list.stream().anyMatch(e -> name.equals(e.getName()));
+    }
+
+    private int getIdFromJson(String json) throws JSONException {
+        return new JSONObject(json).getInt("id");
+    }
+
+    private void checkIfUserGroupExists(int groupId) throws IOException {
+        getUserGroupById(groupId);
+    }
+
+    private <T extends BaseEntity> T resultResolver(Class<T> type, String returnValue) throws IOException {
+        if (returnValue.startsWith(AUTH_TYPE) || returnValue.startsWith(CREATED_AT)) {
+            return new Gson().fromJson(returnValue, type);
+        }
+        if (returnValue.startsWith(MESSAGE_USER_DOES_NOT_EXIST) || returnValue.equals(MESSAGE_INTERNAL_SERVER_ERROR)) {
+            throw new IllegalArgumentException(returnValue);
+        }
+        throw new IOException(returnValue);
+    }
+
+    private String post(String url, String json) throws IOException {
+        return post(url, json, false);
+    }
+
+    private String get(String url) throws IOException {
+        return get(url, true);
+    }
+
+    private String delete(String url) throws IOException {
+        return delete(url, false);
+    }
+
+    private String post(String url, String json, boolean checkResponseStatus) throws IOException {
+        RequestBody body = RequestBody.create(JSON, json);
+        Request request = new Request.Builder()
+                .headers(this.headers)
```

Comment:
Please reorder this to url -> post -> headers (just to be consistent with the other HTTP methods)